audio/wave_shaper_node.rs

```diff
@@ -0,0 +1,116 @@
+use block::Chunk;
+use node::{AudioNodeEngine, BlockInfo};
+use node::{AudioNodeType, ChannelInfo};
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum OverSampleType {
+    None,
+    Double,
+    Quadruple,
+}
+
+type WaveShaperCurve = Option<Vec<f32>>;
+
+#[derive(Clone, Debug)]
+pub struct WaveShaperNodeOptions {
+    pub curve: WaveShaperCurve,
+    pub oversample: OverSampleType,
+}
+
+impl Default for WaveShaperNodeOptions {
+    fn default() -> Self {
+        WaveShaperNodeOptions {
+            curve: None,
+            oversample: OverSampleType::None,
+        }
+    }
+}
+
+#[derive(Clone, Debug)]
+pub enum WaveShaperNodeMessage {
+    SetCurve(WaveShaperCurve),
+}
+
+#[derive(AudioNodeCommon)]
+pub(crate) struct WaveShaperNode {
+    curve_set: bool,
+    curve: WaveShaperCurve,
+    oversample: OverSampleType,
```

Comment:
Let's remove the build time warning and leave a TODO here with a link to a new issue, please.
```rust
#[allow(dead_code)]
// TODO implement tail-time based on the oversample attribute.
// https://github.com/servo/media/issues/XXX
oversample: OverSampleType,
```